src/main/java/org/zalando/nakadi/validation/schema/PartitionKeyFieldsConstraint.java

```diff
@@ -0,0 +1,17 @@
+package org.zalando.nakadi.validation.schema;
+
+import org.zalando.nakadi.domain.EventType;
+import org.zalando.nakadi.domain.EventTypeBase;
+
+import java.util.Optional;
+
+public class PartitionKeyFieldsConstraint implements SchemaEvolutionConstraint {
```

Comment:
@v-stepanov I wasn't sure about the classification of this constraint. One could say that it's about the event type not about the schema, but it references schema properties, so I considered it to be a schema constraint.

The reason why it's not allowed to be changed is to not break order guarantees.